### PR TITLE
CDAP-4689 Use ReflectionUtils in MultipleOutputs, so that setConf is …

### PR DESCRIPTION
**Pull request by anwar6953**
Use ReflectionUtils in MultipleOutputs, so that setConf is called on the instantiated class if its Configurable.
Use OutputFormatClass's ClassLoader when instantiating the OutputFormat.
Additional details in the JIRA: https://issues.cask.co/browse/CDAP-4689

http://builds.cask.co/browse/CDAP-RBT617-2

### CODE REVIEW

**Comment by anwar6953** [comment]
Tested on SDK + Cluster.

File: cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/batch/dataset/MultipleOutputs.java [hunk 1]
```diff
@@ -185,16 +183,24 @@ private synchronized RecordWriter getRecordWriter(String namedOutput) throws IOE
...
+      }
 
-        Instantiator<? extends OutputFormat<?, ?>> instantiator =
-          new InstantiatorFactory(false).get(TypeToken.of(taskContext.getOutputFormatClass()));
-        OutputFormat<?, ?> outputFormat = instantiator.create();
+      ClassLoader outputFormatClassLoader = outputFormatClass.getClassLoader();
+      ClassLoader oldClassLoader = ClassLoaders.setContextClassLoader(outputFormatClassLoader);
 
+      try {
+        OutputFormat<?, ?> outputFormat =
```
Comment (albertshau) [inline cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/batch/dataset/MultipleOutputs.java]: can you add some comments on why we need to use ReflectionUtils

File: cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/batch/dataset/MultipleOutputs.java [hunk 2]
```diff
@@ -185,16 +183,24 @@ private synchronized RecordWriter getRecordWriter(String namedOutput) throws IOE
...
+        outputFormatClass = taskContext.getOutputFormatClass();
+      } catch (ClassNotFoundException e) {
+        throw new IOException(e);
+      }
 
-        Instantiator<? extends OutputFormat<?, ?>> instantiator =
-          new InstantiatorFactory(false).get(TypeToken.of(taskContext.getOutputFormatClass()));
-        OutputFormat<?, ?> outputFormat = instantiator.create();
+      ClassLoader outputFormatClassLoader = outputFormatClass.getClassLoader();
+      ClassLoader oldClassLoader = ClassLoaders.setContextClassLoader(outputFormatClassLoader);
```
Comment (albertshau) [inline cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/batch/dataset/MultipleOutputs.java]: can you add some comments about why this is required

**Comment by albertshau** [comment]
just comments about adding comments. Otherwise lgtm

**Comment by anwar6953** [comment]
Added comments to lines 196 and 201.

**Comment by albertshau** [comment]
:+1:

**Pull request merged**